backend/dataall/modules/dataset_sharing/cdk/pivot_role_data_sharing_policy.py

```diff
@@ -16,7 +16,14 @@ def get_statements(self):
                 effect=iam.Effect.ALLOW,
                 actions=[
                     'iam:PutRolePolicy',
-                    'iam:DeleteRolePolicy'
+                    'iam:DeleteRolePolicy',
```

Comment:
Based on the way we name the managed policies could we drill down the resources to something like `arn:aws:iam::aws:policy/{resource_prefix}-env-{environment_uri}-share-*` ?

If so we then may not have to add any additional excpetions to `.checkov.baseline` too

Reply:
AttachPolicy permissions should be set for the consumption role. Name of the consumption role can be random

Reply:
What we can do is restrict the Create/DeletePolicy, Create/DeletePolicyVersion

Reply:
On second thought we will have to also include delete policy permissions for the inline policies we are removing, 
`dataall-targetDatasetS3Bucket-AccessControlPolicy` and `targetDatasetAccessControlPolicy`

Reply:
I have scoped down the permissions for the first case but not for the old inline policies, I will test with what we ahve now and see how the backwards comaptibility works